collect_app/src/main/java/org/odk/collect/android/backgroundwork/AutoUpdateTaskSpec.java

```diff
@@ -86,7 +87,7 @@
                                         formDownloader.downloadForm(serverFormDetails, null, null);
                                         results.put(serverFormDetails, TranslationHandler.getString(context, R.string.success));
                                     } catch (FormDownloadException e) {
-                                        results.put(serverFormDetails, TranslationHandler.getString(context, R.string.failure));
+                                        results.put(serverFormDetails, new FormDownloadExceptionMapper(context).getMessage(e));
```

Comment:
Note: there isn't an easy way to show this message in "match exactly" mode as far as I can tell because the `FormDownloadException` gets converted to a generic `FormSourceException`. I think those two should probably be consolidated at some point but now didn't feel like the right time.